File: AtCoder/ABC/B/271/B.rb
n, q = gets.chomp.split.map(&:to_i)
a = Array.new
n.times do 
    a << gets.chomp.split.map(&:to_i)[1..]
end
q.times do
    s, t = gets.chomp.split.map(&:to_i) 
    p a[s-1][t-1]
end

File: AtCoder/ABC/B/271/B.py
n, q = map(int, input().split())
a = list()
for i in range(n):
    a.append(list(map(int, input().split()))[1:])
for i in range(q):
    s, t = map(int, input().split())
    print(a[s-1][t-1])